### ISSUE TITLE
Upgrading analyzers throws null reference exceptions

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#629]
The starting point for this bug is a project that has a NuGet entry for Microsoft.CodeAnalysis.CSharp.Workspaces at version 1.0.0-beta1-20141031-01.  In the Packaga Manager Console issue the following command

> update-package -pre -project Microsoft.DotNet.CodeFormatting Microsoft.CodeAnalysis.CSharp.Workspaces

That will result in the following errors during the update while adding a new reference to Microsoft.CodeAnalysis.Analyzers package (was not present before the update):

```
Successfully added 'Microsoft.CodeAnalysis.Analyzers 1.0.0-rc1' to Microsoft.DotNet.CodeFormatting.
You cannot call a method on a null-valued expression.
At C:\Users\jaredpar\Documents\GitHub\codeformatter\src\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\Users\jaredpar\Documents\GitHub\codeformatter\src\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\Users\jaredpar\Documents\GitHub\codeformatter\src\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:8 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
You cannot call a method on a null-valued expression.
At C:\Users\jaredpar\Documents\GitHub\codeformatter\src\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0-rc1\tools\install.ps1:17 char:5
+     $project.Object.AnalyzerReferences.Add($analyzerFilePath.FullName)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
```

**Comment by jaredpar** [comment]
Here is a pre-canned repro for this bug:

- `git clone https://github.com/dotnet/codeformatter.git`
- `git checkout -b repro 8479c0f3779dc98da2175366d3c02af2ba557518`
- Open solution in Visual Studio 2013 
- Open Package Manager Console
- `update-package -pre -project Microsoft.DotNet.CodeFormatting Microsoft.CodeAnalysis.Workspaces.CSharp `

**Comment by srivatsn** [comment]
Ah I see what's happening. The install.ps1 is calling the AddAnalyzerReference DTE API to add the analyzer but on VS2013 that API doesn't exist. We should add a null-check so that when analyzer packages are installed in 2013, it becomes a no-op.

**Comment by jmarolf** [comment]
@jaredpar this was fixed with an internal changeset 1424664. Could you please verify?

**Comment by lgolding** [comment]
Assigning to Sri since he already has 2013 installed.
